os/arch/arm/src/armv7-a/arm_gicv2.c

```diff
@@ -508,6 +508,43 @@ int up_prioritize_irq(int irq, int priority)
 	return -EINVAL;
 }
 
+/****************************************************************************
+ * Name: up_affinity_irq
+ *
+ * Description:
+ *   set the affinity of a given irq (SPI only) to given cpu.
+ *   Currenlty, it doesnot support assigning multiple cpus as affinity.
+ *   It can assign only one cpu per SPI.
+ *
+ * Input Parameters:
+ *   irq: irq id
+ *   cpu: cpu id
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void up_affinity_irq(int irq, int cpu)
+{
+	DEBUGASSERT(irq >= GIC_IRQ_SPI && irq < NR_IRQS && cpu >= 0 && cpu < CONFIG_SMP_NCPUS);
+
+	int target_cpu_bit = 1 << cpu;
+
+	/* read the current cpu affinity register */
+	uint32_t reg_value = getreg32(GIC_ICDIPTR(irq));
+
+	/* clear the bits for the current irq */
+	reg_value &= ~(0xFF << ((irq % 4) * 8));
+
+	/* write the target core for the irq */
+	reg_value |= (target_cpu_bit << ((irq % 4) * 8));
+
+	/* write the updated value back to the register */
+	putreg32(reg_value, GIC_ICDIPTR(irq));
+
+}
+
```

Comment:
Let us add another api `up_reset_irq_affinity`. This will set the irq affinity to core 0 only. This is the default setting.